common/coin_support/coin_utils.h

```diff
@@ -108,17 +108,50 @@ typedef struct
     uint8_t change_count[1];
     address_type *change;
 
-    uint8_t transactionFees[4];
+    uint8_t transaction_fees[8];
 
     uint8_t decimal[1];
 
-    char token_name[8];
+    char *token_name;
 
     uint8_t network_chain_id;
 
 } txn_metadata;
 #pragma pack(pop)
 
+#pragma pack(push, 1)
+/**
+ * @brief Stores the deserialized information from desktop for the receive transaction process.
+ * @details
+ *
+ * @see receive_transaction_controller(), receive_transaction_controller_eth(), receive_transaction_tasks(),
+ * desktop_listener_task(), RECV_TXN_START
+ * @since v1.0.0
+ */
+typedef struct Receive_Transaction_Data {
+  uint8_t wallet_id[WALLET_ID_SIZE];
+  uint8_t purpose[4];
+  uint8_t coin_index[4];
+  uint8_t account_index[4];
+  uint8_t chain_index[4];
+  uint8_t address_index[4];
+  char *token_name;
+  union {
+    uint8_t network_chain_id;
```

Comment:
We will add new EVM chains. Would it be good to expand this now? Chain IDs can be multi-byte value https://chainlist.org/.

If we do this, we can remove the union.

Reply:
We can do this, however, this would delay the build as it requires changes from both sides.

Reply:
We will take this up in later discussions